node/resolvers/profile/index.ts

```diff
@@ -44,6 +52,38 @@ export const mutations = {
 export const queries = {
   profile: (_: any, { customFields }: any, context: Context) =>
     getProfile(context, customFields),
+
+  checkProfileAllowed: async (_: any, __: any, context: Context) => {
+    const {
+      clients: { catalog, session },
+      vtex: { segment },
+      cookies,
+    } = context
+    const salesChannel = segment ? segment.channel : null
+
+    const { sessionData } = await session.getSession(
+      cookies.get('vtex_session')!,
+      ['*']
+    )
+
+    const email: string | undefined = path(
+      ['namespaces', 'profile', 'email', 'value'],
+      sessionData
+    )
+
+    if (!email) {
+      return { allowed: false }
+    }
+
+    const availableSalesChannels = await catalog
+      .salesChannelAvailable(email)
+      .catch(() => [])
+
+    // Checking with `==` since `sc.Id` is an Integer and salesChannel a string
+    const available = availableSalesChannels.find(sc => sc.Id == salesChannel)
```

Comment:
or convert sc.Id to toString() and use ===. anyway 🤷‍♀

Reply:
🤷‍♂